app/scripts/services/dimStoreService.factory.js

```diff
@@ -1459,6 +1441,32 @@
       }
     }
 
+    function setItemYear(item) {
+      // determine what year this item came from based on sourceHash value
+      // items will hopefully be tagged as follows
+      // No value: Vanilla, Crota's End, House of Wolves
+      // The Taken King (year 2): 460228854
+      // Rise of Iron (year3): 24296771
+
+      // This could be further refined for CE/HoW based on activity. See
+      // DestinyRewardSourceDefinition and filter on %SOURCE%
+      // if sourceHash doesn't contain these values, we assume they came from
+      // year 1
+      item.sourceHashes = item.sourceHashes || [];
```

Comment:
This'll break things - we need to keep this up top, so that createdItem.sourceHashes is never null.